### ISSUE TITLE
cooja_mthread: location of data pointer on stack does not seem to be correct (x86_64)

### ISSUE THREAD

**Issue by marcas756** [contiki-os/contiki#1007]
I am trying to get contiki mtarch running on platform cooja. First attempt was to get it running with the platform/native implementation ( #992) using win fibers/ linux ucontext , with success, but maybe not covering all OSs on x86/x64, also there is an cooja_mtarch implementation used in platform cooja which may be of use for me.

I rewrote contiki-cooja-main.c a little bit to check if the cooja_mtarch implementation is able to handle a thread properly, but it fails:

```C
const char *teststring = "Teststring";

JNIEXPORT void JNICALL
Java_se_sics_cooja_corecomm_CLASSNAME_init(JNIEnv *env, jobject obj)
{
  /* Create rtimers and Contiki threads */
  cooja_mt_start(&rtimer_thread, &rtimer_thread_loop, NULL);
  cooja_mt_start(&process_run_thread, &process_run_thread_loop, teststring);
 }
```

```C
static void
process_run_thread_loop(void *data)
{
    printf("process_run_thread_loop data : %p, %s \n",data,(char*)data);
    /* Yield once during bootup */
    simProcessRunValue = 1;
    cooja_mt_yield();
...
```

00:00.381	ID:1	process_run_thread_loop data : **(nil), (null)**
00:00.382	ID:1	Contiki-2.7 started. Node id is set to 1.
00:00.382	ID:1	Rime started with address 1.0
00:00.382	ID:1	MAC 01:00:00:00:00:00:00:00 Rime/CSMA/nullrdc, channel check rate 1000 Hz
00:00.382	ID:1	Starting 'Hello world process'
00:00.382	ID:1	Hello, world

I also made some tests with cpu/x86 (where platform/cooja/sys/cooja_mtarch is derived from) ... with similar behaviour, also if the two implementations are  not the same (see sp position in struct).

Branch: 
contiki-2-7

System:
```
Linux dualcore 3.13.0-48-generic #80-Ubuntu SMP Thu Mar 12 11:16:15 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by marcas756** [comment]
But it seems to work for 32 Bit architecture:

```
Linux instant-contiki 3.2.0-51-generic-pae #77-Ubuntu SMP Wed Jul 24 20:40:32 UTC 2013 i686 athlon i386 GNU/Linux
```

00:00.381	ID:1	process_run_thread_loop data : TestString
00:00.382	ID:1	Contiki-2.6-900-ga6227e1 started. Node id is set to 1.
00:00.382	ID:1	Rime started with address 1.0
00:00.382	ID:1	MAC 01:00:00:00:00:00:00:00 Rime/CSMA/nullrdc, channel check rate 1000 Hz
00:00.382	ID:1	Starting 'Hello world process'
00:00.382	ID:1	Hello, world

**Comment by marcas756** [comment]
Fixed via #1012
